client/lib/wp/sync-handler/index.js

```diff
@@ -187,13 +193,48 @@ export class SyncHandler {
 		return localforage.getItem( key, fn );
 	}
 
+	/**
+	 * Add/Override a record.
+	 *
+	 * @param {String} key - record key identifier
+	 * @param {Object} data - data to store
+	 * @param {Function} [fn] - callback function
+	 * @return {Promise} natuve ES6 promise
+	 */
 	storeRecord( key, data, fn = () => {} ) {
 		debug( 'storing data in %o key\n', key );
-		return localforage.setItem( key, data, fn );
+
+		// add this record to history
+		return cacheInvalidation
```

Comment:
Perhaps the name `cacheInvalidation` is throwing us off. I was suggesting to abstract out the localforage details in the cache. Does the following still look weird?
```javascript
const cache = new MaxAgeCache();
cache.addItem( key, data );
cache.removeItem( key, data);
cache.clearAll();
```